src/error.rs

```diff
@@ -0,0 +1,33 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum AstarteMessageHubError {
+    #[error("")]
+    Infallible(#[from] std::convert::Infallible),
+
+    #[error("")]
+    TryFromIntError(#[from] core::num::TryFromIntError),
```

Comment:
Why a specific error for ints?

Reply:
It is used to propagate  the `conversion error` from timestamp to chrono::datetime
https://docs.rs/pbjson-types/0.5.1/pbjson_types/struct.Timestamp.html#associatedtype.Error